backend/src/api/blocks.ts

```diff
@@ -932,7 +932,39 @@ class Blocks {
         }
 
         await blocksRepository.$saveBlockInDatabase(blockExtended);
-        this.updateTimerProgress(timer, `saved ${this.currentBlockHeight} to database`);
+        this.updateTimerProgress(
```

Comment:
My initial comment is to try to build the code in such a way that merging up/down stream is easy as possible

Reply:
fixed